src/ansys/fluent/core/solver/flobject.py

```diff
@@ -456,10 +457,31 @@ def __getattribute__(self, name):
         if name in super().__getattribute__("child_names"):
             if not self.is_active():
                 raise RuntimeError(f"'{self.path}' is currently not active")
-        return super().__getattribute__(name)
+        try:
+            return super().__getattribute__(name)
+        except AttributeError as ex:
+            parents = []
+            for parent in self.get_active_child_names():
+                try:
+                    if hasattr(getattr(self, parent), str(name)):
+                        parents.append(parent)
+                except AttributeError:
+                    pass
+            if len(parents):
+                print(f"\n {name} is a child of {parents} \n")
+            raise AttributeError(
+                allowed_name_error_message(
+                    "Settings objects", name, super().__getattribute__("child_names")
+                )
+            ) from ex
```

Comment:
This except block is quite complicated and the reader needs to scrutinize it in some detail to see what is going on - that's bad for maintenance.
Better in general:
```
try:
    do_stuff()
except ParticularError as ex:
    handle_particular_error_in_explicitly_described_way(ex, args)
```
 
- Note that `handle_particular_error_in_explicitly_described_way` can now also be tested in isolation.
- By isolating the handling in a single function, it can easily be switched off

Reply:
Done. Thank you.